GearWorksXO-2017/src/org/usfirst/frc/team2212/subsystems/Lift.java

```diff
@@ -0,0 +1,82 @@
+package org.usfirst.frc.team2212.subsystems;
+
+import java.util.function.Supplier;
+
+import com.spikes2212.dashboard.ConstantHandler;
+import com.spikes2212.genericsubsystems.LimitedSubsystem;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.PIDSource;
+import edu.wpi.first.wpilibj.SpeedController;
+
+/**
+ * @author Itamar Rivkind
+ *
+ */
+
+public class Lift extends LimitedSubsystem {
+
+	private SpeedController motor;
+	private DigitalInput upLimit;
+	private DigitalInput downLimit;
+	private Encoder encoder;
+	public static final Supplier<Double> SPEED = ConstantHandler.addConstantDouble("Lift-SPEED", 0.7);
+	public static /* final */ Supplier<Integer> MIDDLE; // place in encoder for
+														// putting lower gear
```

Comment:
change MIDDLE to MIDDLE_SET_POINT so it will be different than the MIDDLE in the enum